plugins/woocommerce/tests/e2e-pw/global-setup.js

```diff
@@ -178,6 +181,62 @@ module.exports = async ( config ) => {
 		process.exit( 1 );
 	}
 
+
+	// While we're here, let's set HPOS according to the passed in ENABLE_HPOS env variable
+	// This was always being set to 'yes' after login in wp-env so this step ensures the
+	// correct value is set before we begin our tests
+	const hposSettingRetries = 5;
+	const api = new wcApi( {
+		url: baseURL,
+		consumerKey: process.env.CONSUMER_KEY,
+		consumerSecret: process.env.CONSUMER_SECRET,
+		version: 'wc/v3',
+	} );
+
+	for ( let i = 0; i < hposSettingRetries; i++ ) {
+		try {
+			if (ENABLE_HPOS === undefined || ENABLE_HPOS === '0') {
+				console.log( 'Trying to switch off HPOS...' );
+				// call API to update HPOS setting
+				const HPOSSetOffResponse = await api.post( 'settings/advanced/woocommerce_custom_orders_table_enabled', {
+					value: "no"
+				} );
+				// validate HPOS setting
+				if (HPOSSetOffResponse.data.value === 'no') {
+					console.log('HPOS Switched off successfully');
+					hposConfigured = true;
+					break;
+				}
+			}
+			if (ENABLE_HPOS === '1') {
+				console.log( 'Trying to switch on HPOS...' );
+				// call API to update HPOS setting
+				const HPOSSetOnResponse = await api.post( 'settings/advanced/woocommerce_custom_orders_table_enabled', {
+					value: "yes"
+				} );
+				// validate HPOS setting
+				if (HPOSSetOnResponse.data.value === 'yes') {
+					console.log('HPOS Switched on successfully');
+					hposConfigured = true;
+					break;
+				}
+			}
+			break;
+		} catch ( e ) {
+			console.log(
+				`HPOS setup failed. Retrying... ${ i }/${ hposSettingRetries }`
+			);
+			console.log( e );
+		}
+	}
```

Comment:
I think we can simplify this code if we get rid of the `if` structures and make better use of the `ENABLE_HPOS` variable. Something in line with:

```
const value = ENABLE_HPOS === '1' ? 'yes' : 'no';

for (let i = 0; i < hposSettingRetries; i++) {
    try {
      console.log( `Trying to switch ${ value === 'yes' ? 'on' : 'off' } HPOS...` );
      const response = await api.post( 'settings/advanced/woocommerce_custom_orders_table_enabled', { value } );
      if ( response.data.value === value ) {
        console.log( `HPOS Switched ${ value === 'yes' ? 'on' : 'off' } successfully` );
        hposConfigured = true;
        break;
      }
    } catch (e) {
      console.log( `HPOS setup failed. Retrying... ${ i }/${ hposSettingRetries }` );
      console.log(e);
    }
  }
```

Feel free to amend any variable name if you decide to follow this approach!

Reply:
➕ 1️⃣  ☝️

Reply:
> I think we can simplify this code if we get rid of the `if` structures and make better use of the `ENABLE_HPOS` variable. Something in line with:
> 
> ```
> const value = ENABLE_HPOS === '1' ? 'yes' : 'no';
> 
> for (let i = 0; i < hposSettingRetries; i++) {
>     try {
>       console.log( `Trying to switch ${ value === 'yes' ? 'on' : 'off' } HPOS...` );
>       const response = await api.post( 'settings/advanced/woocommerce_custom_orders_table_enabled', { value } );
>       if ( response.data.value === value ) {
>         console.log( `HPOS Switched ${ value === 'yes' ? 'on' : 'off' } successfully` );
>         hposConfigured = true;
>         break;
>       }
>     } catch (e) {
>       console.log( `HPOS setup failed. Retrying... ${ i }/${ hposSettingRetries }` );
>       console.log(e);
>     }
>   }
> ```
> 
> Feel free to amend any variable name if you decide to follow this approach!

I like it - thanks @alopezari and @rodelgc !

Reply:
One thing to note is that based on:

> const value = ENABLE_HPOS === '1' ? 'yes' : 'no';

any ENABLE_HPOS value that is not '1' will set HPOS to off

Reply:
Resolving this convo now.